src/edu/csus/ecs/pc2/clics/API202306/ScoreboardService.java

```diff
@@ -0,0 +1,138 @@
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getScoreboard(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId,
+            @QueryParam("group_id") String group_id, @QueryParam("division") String division) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == true) {
+            ContestTime contestTime = model.getContestTime();
+            // verify contest has started or user is special
+            if (contestTime.getElapsedMS() > 0 ||
+                ((sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) ||
+                  sc.isUserInRole(WebServer.WEBAPI_ROLE_ANALYST) ||
+                  sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE)))) {
+
+                Group specificGroup = null;
+                Integer divNumber = null;
+
+                // if a specific group was requested, let's look for that so we can pass it to the standings routine
+                if(!StringUtilities.isEmpty(group_id)) {
+                    for(Group group: model.getGroups()) {
+                        if (group.isDisplayOnScoreboard() && IJSONTool.getGroupId(group).equals(group_id)) {
+                            specificGroup = group;
+                            break;
+                        }
+                    }
+                    if(specificGroup == null) {
+                        return Response.status(Response.Status.NOT_FOUND).build();
+                    }
+                }
+
+                // see if a division number was specified - this is a PC2 extension(!!) and is NOT part of the CLICS API, but,
+                // we are allowed to add things.
+                if(!StringUtilities.isEmpty(division)) {
+                    try {
+                        divNumber = Integer.parseInt(division);
+                    } catch(Exception e) {
+                        // Bad division specified
+                        return Response.status(Response.Status.NOT_FOUND).build();
+                    }
+                }
+
+                // ok to return scoreboard
+                try {
+                    CLICSScoreboard scoreboard = new CLICSScoreboard(model, controller, specificGroup, divNumber);
+                    return Response.ok(scoreboard.toJSON(), MediaType.APPLICATION_JSON).build();
+                } catch (IllegalContestState | JAXBException | IOException e) {
+                    controller.getLog().log(Log.WARNING, "Exception creating PC2 scoreboard JSON: " + e.getMessage(), e);
+                    return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+                }
+
+            } else {
+                // do not show (return) the scoreboard if the contest has not
+                // been started and the requester is not special)
+                // FIXME: might be better if this returned an "empty" scoreboard?
```

Comment:
To me, returning FORBIDDEN (as it currently does) is better than returning an empty scoreboard.  An empty scoreboard sort of says "there is no scoreboard", whereas FORBIDDEN says "Yes, there's a scoreboard -- but you're not allowed to see it".

Reply:
Agree. Removed the FIXME.  Next push.